backend/src/src-licenses/src/main/java/com/siemens/sw360/licenses/db/LicenseDatabaseHandler.java

```diff
@@ -370,7 +372,10 @@ public RequestStatus updateLicense(License inputLicense, User user, User request
 
             License dbLicense = new License();
             if(inputLicense.isSetId()) {
-                dbLicense = licenseRepository.get(inputLicense.getId());
+                License dbLicenseTmp = licenseRepository.get(inputLicense.getId());
```

Comment:
Well, at the moment it works because in the frontend, the id is not set. But someone could have the idea to set the id in the frontend, because you know that it is equal to the shortname. Then, the id would be set and the license would not be in the database. Then you get null and a null pointer exception thereafter, when you call the updateLicenseFromInputLicense method.

And - yes - some lines later, we should modify the test whether to add or to update in the same way.

Actually we should make the license id a required field, now that the shortname is gone. And then you should/have to set the id in the Portlet already.